entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -224,6 +224,9 @@ public boolean equals(Object o) {
             return false;
         }
         <%= entityClass %> <%= entityInstance %> = (<%= entityClass %>) o;
+        if(<%= entityInstance %>.id == null || id == null) {
```

Comment:
Doesn't it break a contract?
Two objects with null IDs should be equal if ID is a part of equals and hashcode. With this implementation it is false.